### ISSUE TITLE
Makefile cleanups and gcc

### ISSUE THREAD

**Issue by coldacid** [MrZammler/iGame#92]
I've been working on some changes with the makefiles for iGame, related both to the OS4 build and the `release` target, to bring things more in-line with the Docker makefile used by @walkero-gr for his CI builds. My only real issue is with the basic `Makefile` file, since it still uses GCC instead of VBCC.

Is anyone actually using the GCC-based `Makefile` or should I just remove it with this work?

BTW @walkero-gr I've left `Makefile.docker` alone in this work. Should I take your changes to it for your CI build branch and add it to my branch before I PR any of this?

**Comment by walkero-gr** [comment]
I totally agree about the Makefiles cleanup. I believe that some of them might not be needed. I don't use the GCC one, but @MrZammler does.

Also, we can have Windows, MacOS and Linux all merged into one Makefile by using ENV variables. And maybe docker makefile can be merged there as well. This is a good to not have specific paths in the Makefile and let the user set his own in ENV variables. That's how the Makefile.docker works. We just need to add some good documentation in the wiki for that.

What do you think?

**Comment by coldacid** [comment]
Well, we don't even have a MacOS makefile. I think `Makefile.linux` would actually work just fine on MacOS so long as all the required tools are installed and on the path. As for the Linux and Windows makefiles, they're getting closer but I'm not sure if we're at the point where we can merge them together yet.

I'm not sure if the GCC makefile is even used by @MrZammler because the flags it provides to GCC seem to be the same as for VBCC, which doesn't work at all. I've tried different things to try and make GCC builds work locally, but even when the proper flags are provided, include files seem to keep doing things that GCC doesn't like, causing compilation to fail.

For merging the Docker makefiles with the others, I think we first need to get the CI stuff mainlined.

**Comment by coldacid** [comment]
@MrZammler @walkero-gr could I get a review on #111 and any feedback on the effort in general?

**Comment by javierdlr** [comment]
I just froked iGame and created an AmigaOS'ware GCC makefile:

https://github.com/javierdlr/iGame/commit/363daefd0a4fb000f76c4874af0341c17a82c68a
